daffodil-runtime1/src/main/scala/org/apache/daffodil/core/compiler/Compiler.scala

```diff
@@ -31,17 +31,21 @@ import scala.collection.mutable.ArrayBuffer
 import scala.util.Try
 import scala.xml.Node
 
+import org.apache.daffodil.api.compiler.{ ProcessorFactory => JProcessorFactory }
+import org.apache.daffodil.api.exceptions.{ InvalidParserException => JInvalidParserException }
+import org.apache.daffodil.api.{ CodeGenerator => JCodeGenerator }
+import org.apache.daffodil.api.{ DataProcessor => JDataProcessor }
```

Comment:
2 things:

1. Just to make sure I'm tracking the API changes here correctly, we are dropping having separate Scala and Java APIs and instead having a single unified API that is Java compatible?
2. Assuming 1 is correct do we really want to be prefixing everything with "J"?  It made sense when there was separate APIs but unless there is some sort of loss of functionality when using the new API from Scala.  I'm open to ideas on this but just thought it might not be necessary.

Reply:
Re 2, The J prefix is because alot of the classes share the same name as their Scala counterpart ex InfosetInputter, DataProcessor, so we needed to rename them and J seemed like the best option. Any suggestions?

Re 1, we are also integrating the new JAVA API into our current backend for compatibility @stevedlawrence is that right?

Reply:
>   we needed to rename them and J seemed like the best option. Any suggestions?

One alternative could be to just `import org.apache.daffodil.api` and then classes could do something like
```scala
class DataProcessor extends api.DataProcessor {
...
}
```
Possibly removes some imports and makes also makes clear that thigns are implementing the API interface.

> Re 1, we are also integrating the new JAVA API into our current backend for compatibility @stevedlawrence is that right?

Yes, we will now just have a single API that works in both Java and Scala. Forease of use, it probably doesn't want to reference any Scala-types. This should make it easy to use for Java people, while still being easy to use for scala users.

Reply:
Ok, so the Java version of the InfosetInputter doesn't contain all the actual functionality, it just exposes the necessary function calls for external use?  I wasn't sure if we had actually fully reimplemented things like the InfosetInputter in Java, in which case I didn't see the need for a J prefix.  If we aren't doing that and there are separate Scala versions of these classes, then the J prefix makes sense.